pkg/jobconfig/files_test.go

```diff
@@ -808,8 +816,216 @@ func TestInfo_ConfigMapName(t *testing.T) {
 	for _, testCase := range testCases {
 		t.Run(testCase.expected, func(t *testing.T) {
 			info := Info{Branch: testCase.branch, Type: testCase.jobType}
-			if actual, expected := info.ConfigMapName(), testCase.expected; !reflect.DeepEqual(actual, expected) {
-				t.Errorf("%s: didn't get correct basename: %v", testCase.name, diff.ObjectReflectDiff(actual, expected))
+			if diff := cmp.Diff(testCase.expected, info.ConfigMapName()); diff != "" {
+				t.Errorf("%s: didn't get correct basename: %v", testCase.name, diff)
+			}
+		})
+	}
+}
+
+func TestPrune(t *testing.T) {
+	testCases := []struct {
+		name            string
+		jobconfig       *prowconfig.JobConfig
+		expectedPruned  bool
+		generatedLabels []string
+	}{
+		{
+			name: "stale generated presubmit is pruned",
```

Comment:
nit: test periodics?

Reply:
If you mean that I should also test periodics, I have one toward the bottom now.